src/commands/suspend.js

```diff
@@ -0,0 +1,96 @@
+const Discord = require('discord.js');
+const getConfig = require('../config/get-config.js');
+const config = getConfig();
+module.exports = {
+  prefix: 'suspend',
+  description: 'Suspends a user for the given reason',
+  command: async function suspend(message) {
+    //check for appropriate permissions
+    if (!message.member.hasPermission('KICK_MEMBERS')) {
+      console.log(
+        `${message.author.username} did not have the correct permissions.`
+      );
+      return;
+    }
+    //check for log channel setting
+    const modChannel = message.guild.channels.cache.find(
+      (channel) => channel.name === config.LOG_MSG_CHANNEL
+    );
+    if (!modChannel) {
+      console.log('Log channel not found.');
+      return;
+    }
+    //check for suspend category setting
+    const suspendCategory = config.SUSPEND_CATEGORY;
+    const category = await message.guild.channels.cache.find(
+      (c) => c.name === suspendCategory && c.type === 'category'
+    );
+    if (!category) {
+      console.log('Missing suspend category.');
+      return;
+    }
+    //check for suspend role setting
+    const suspend = message.guild.roles.cache.find(
+      (role) => role.name == config.SUSPEND_ROLE
+    );
+    if (!suspend) {
+      console.log(`Missing suspend role.`);
+      return;
+    }
+
+    const mod = message.author;
+    const msgArguments = message.content.split(' ');
+    const user = message.mentions.members.first();
+    //check for valid user tag
+    if (!user) {
+      message.channel.send(`Invalid user tag.`);
+      return;
+    }
+    //cannot target self
+    if (message.mentions.users.first() === mod) {
+      message.channel.send(`Cannot target self.`);
+      return;
+    }
+    const reasonArg = msgArguments.slice(3, msgArguments.length);
+    //check for reason provided, if none then create one.
+    const reason = reasonArg.join(' ') || 'No reason provided';
+    //logging embed
+    const restrictEmbed = new Discord.MessageEmbed()
+      .setColor('#FF0000')
+      .setTitle(`Access Restricted!`)
+      .addFields(
+        {
+          name: 'What happened?',
+          value: `${mod} has suspended ${user}.`
+        },
+        {
+          name: 'Reason',
+          value: `${reason}`
+        }
+      )
+      .setFooter('Please remember to follow our rules!');
+    modChannel.send(restrictEmbed);
+    //assign roles
+    user.roles.remove(user.roles).catch((e) => console.error(e));
+    user.roles.add(suspend).catch((e) => console.error(e));
```

Comment:
Same as the other comment, this can still fail silently in that we will say that we removed/updated the user(s) but they could fail, which is mis-leading. We could say nothing and log the actual problem, or return a message.

Reply:
Realistically, the only reason either this or the above comment should throw any error is if the bot itself doesn't have the correct permissions. The command code checks for everything else and terminates if a configuration is missing or a condition isn't met. I could look at adding a bot permission check if we think that's necessary - otherwise, yeah we could wrap the entire function in a try block and catch the errors all at once.